cmd/cloud/deployment.go

```diff
@@ -869,7 +869,7 @@ func isValidExecutor(executor string) bool {
 
 // isValidCloudProvider returns true for valid CloudProvider values and false if not.
 func isValidCloudProvider(cloudProvider astrocore.SharedClusterCloudProvider) bool {
-	return cloudProvider == astrocore.SharedClusterCloudProviderGcp || cloudProvider == astrocore.SharedClusterCloudProviderAws
+	return cloudProvider == astrocore.SharedClusterCloudProviderGcp || cloudProvider == astrocore.SharedClusterCloudProviderAws || cloudProvider == astrocore.SharedClusterCloudProviderAzure
```

Comment:
IMO, we should drop this check since we already have the same check on the Core API side. We are not gaining much by having duplicated validation, but end up losing much like in this case where Core API supported Azure deployment for a very long time.

Reply:
Thanks @neel-astro !

If we remove this Validation from CLI then in case where user provide `cloudProvider` value other than `aws`, `azure` or `gcp` CLI will fall back to azure but will not throw any exception. We'll have to relay down the error from core-api to user from CLI it seems.

```
 ./astro deployment create -p openshift                                                                       
Current Workspace: Sample Workspace 2

Please specify a name for your Deployment

Deployment name: test

Please select a Region for your Deployment:
 #     REGION         
 1     westeurope     
 2     westus2        
 3     eastus2 


./astro deployment create -p test                                                              
Current Workspace: Sample Workspace 2

Please specify a name for your Deployment

Deployment name: test

Please select a Region for your Deployment:
 #     REGION         
 1     westeurope     
 2     westus2        
 3     eastus2 
```